app/assets/scripts/components/map/legend.js

```diff
@@ -15,48 +15,48 @@ export default function Legend({ parameters, activeParameter }) {
     history.push(`${location.pathname}?parameter=${parameter}`);
   }
 
-  const parameterNames = [...new Set(parameters.map(p => p.name))];
+  //const parameterNames = [...new Set(parameters.map(p => p.name))];
 
   let drop = (
     <Dropdown
       triggerElement="button"
       triggerClassName="button button--primary-unbounded drop__toggle--caret"
       triggerTitle="Show/hide parameter options"
-      triggerText={activeParameter}
+      triggerText={activeParameter.displayName}
     >
       <ul
         role="menu"
         className="drop__menu drop__menu--select"
         style={{ overflowY: `scroll`, maxHeight: `15rem` }}
       >
-        {parameterNames.map((paramName, i) => (
-          <li key={`${paramName}-${i}`}>
+        {parameters.map(param => (
+          <li key={`${param.id}`}>
             <a
               className={c('drop__menu-item', {
-                'drop__menu-item--active': activeParameter === paramName,
+                'drop__menu-item--active': param.id === activeParameter.id,
               })}
               href="#"
-              title={`Show values for ${paramName}`}
+              title={`Show values for ${param.displayName}`}
               data-hook="dropdown:close"
-              onClick={e => onFilterSelect(paramName, e)}
+              onClick={e => onFilterSelect(param.id, e)}
             >
-              <span>{paramName}</span>
+              <span>{param.displayName}</span>
             </a>
           </li>
         ))}
       </ul>
     </Dropdown>
   );
 
-  const scaleStops = generateLegendStops(activeParameter.toLowerCase());
+  const scaleStops = generateLegendStops(activeParameter.name.toLowerCase());
```

Comment:
`Uncaught TypeError: Cannot read property 'toLowerCase' of undefined`

Reply:
I'm still getting this error

Reply:
this was on country page right? @necoline